src/cwe_checker_lib/src/utils/arguments.rs

```diff
@@ -209,16 +209,20 @@ pub fn create_string_stack_arg(size: ByteSize, stack_offset: i64) -> Arg {
     Arg::Stack {
         offset: stack_offset,
         size,
+        data_type: None,
     }
 }
 
 /// Creates a string register parameter given a register name.
 pub fn create_string_register_arg(size: ByteSize, register_name: String) -> Arg {
-    Arg::Register(Variable {
-        name: register_name,
-        size,
-        is_temp: false,
-    })
+    Arg::Register {
+        var: Variable {
+            name: register_name,
+            size,
+            is_temp: false,
+        },
+        data_type: None,
+    }
 }
```

Comment:
Since the function is named create_*string*_register_arg, I would expect it to set the data_type to `Some(Pointer)`.

Reply:
Fixed.